classes/Cart.php

```diff
@@ -449,7 +449,7 @@ public function getAverageProductsTaxRate(&$cartAmountTaxExcluded = null, &$cart
      *
      * @return array|false|mysqli_result|PDOStatement|resource|null Database result
      */
-    public function getCartRules($filter = CartRule::FILTER_ACTION_ALL, $autoAdd = true)
+    public function getCartRulesWithoutContextualValue($filter = CartRule::FILTER_ACTION_ALL, $autoAdd = true)
```

Comment:
I would advise against creating new public functions if they don't need to be public. It's best to leave it protected.